Update dependencies, implement an ad blocker, and create a workflow for cross-platform builds.

nglmercer opened the pull request:
Tauri update: Add blocker added and some APIs implemented. If there's no interest, I'll do the development entirely on my own. Thanks.

nglmercer committed:
feat: update dependencies to latest versions

Updated Tauri API to v1.6.0, Vue to v3.5.22, Vite to v5.2.8, and related development dependencies. Also updated Rust dependencies in Cargo.lock to ensure compatibility and benefit from latest security patches and improvements.

nglmercer committed:
feat: migrate from Tauri v1 to v2 and enhance window controls

Updated all Tauri dependencies from v1 to v2 including @tauri-apps/api and cli, upgraded Vue from 3.5.22 to 3.4.38, and updated Vite and its Vue plugin. Added custom window controls with settings and minimize buttons styled with hover effects and proper Tauri window integration.

nglmercer committed:
refactor: modularize main.rs and extract tray/window logic

Refactor the main.rs file by extracting tray and window management logic into separate modules (tray.rs and window.rs). This improves code organization and maintainability by separating concerns. Also added a panic handler for better error handling and removed the inline JavaScript injection code, which has been moved to a more appropriate location.

nglmercer committed:
feat: enhance window management and tray functionality

Add Tauri commands for window control (minimize, maximize/unmaximize, close) and integrate them with the invoke handler. Update tray icon to use default window icon and improve tooltip display. Configure window settings with hidden title and transparency for a cleaner appearance. Import MouseButton for more efficient tray event handling.

nglmercer committed:
Add CI, lint, and release workflows; migrate to TypeScript

Introduces GitHub Actions workflows for CI, linting, and release automation. Migrates main entry point from JavaScript to TypeScript, adds src/core/api.ts, and updates .gitignore to ignore lock files. Removes bun.lock and plugin-system-design.md, and updates Tauri configuration and Rust backend files to support the new build and workflow setup.

nglmercer committed:
feat: add TypeScript and adblock support with dependencies

Update package.json to add TypeScript and esbuild as dev dependencies for better development tooling. Add adblock functionality in the Rust backend by including the adblock crate and ureq for HTTP requests. Remove commented configuration lines and simplify Cargo.toml structure.

nglmercer committed:
fix fmt

nglmercer committed:
update workflows

nglmercer committed:
update workflows

nglmercer committed:
workflow changes

nglmercer committed:
change workflow

nglmercer committed:
fix workflow

nglmercer committed:
change install clippy

nglmercer committed:
fix build

nglmercer committed:
fix backend test

nglmercer committed:
fix build

nglmercer committed:
Refactor AdBlock engine for improved thread safety

Reworked the AdBlockEngine to store FilterSet directly and instantiate Engine per request, improving thread safety and simplifying state management. Updated JavaScript injection to use a helper for Tauri command invocation. Removed unnecessary Arc/Mutex usage and streamlined filter loading and blocking logic. Also cleaned up the lint workflow by removing Clippy and dependency installation steps.

nglmercer committed:
Add Tauri Shell plugin and AdBlock capability

Integrated the tauri-plugin-shell and tokio dependencies for shell command support. Added adblock capability manifest and updated schemas to reflect new permissions. Modified adblock_plugin.rs and main.rs to register the shell plugin.

nglmercer committed:
Refactor adblock and window event handling code

Improves code readability and consistency in adblock_plugin.rs by reordering imports, formatting, and simplifying logic. Refactors window.rs to use more concise pattern matching for window and run events, reducing nesting and improving maintainability.

nglmercer committed:
format

nglmercer committed:
Add release workflow and documentation

Introduces a multi-platform release workflow in GitHub Actions with support for manual and automatic releases, artifact renaming, and changelog integration. Adds CHANGELOG.md for version history and RELEASE.md with detailed release instructions and best practices.

nglmercer committed:
fix releases workflow

nglmercer committed:
Refactor AdBlock integration and update dependencies

Replaces legacy AdBlock plugin logic with a new Rust/JS hybrid implementation, including a new adblock.js injector and a refactored Rust plugin with improved filter management and caching. Updates Tauri and related dependencies, revises capabilities and security configuration, and enhances system tray with AdBlock controls. Removes obsolete documentation and release files.

nglmercer committed:
update lint

nglmercer committed:
Refactor adblock plugin and enhance cosmetic filtering

Refactored adblock_plugin.rs to simplify state management, remove redundant cosmetic filter handling, and update cosmetic resource API. Updated adblock.js to support new cosmetic filtering logic and improved cache handling. Added get_hidden_class_id_selectors and get_cosmetic_resources endpoints, updated Cargo.toml adblock features, and adjusted capabilities/adblock.json for new API methods.

nglmercer committed:
Refactor adblock logic and improve YouTube ad skipping

Major refactor of adblock.js to modularize configuration, improve cache management, and enhance dynamic cosmetic filtering. Added robust YouTube ad skipper with overlay hiding and auto-skip logic. Updated adblock_plugin copy.rs to simplify AdBlockState, accept full URLs for cosmetic resources, and handle exceptions for hidden selectors. Improved error handling and serialization in Rust backend.

nglmercer committed:
add sample

nglmercer committed:
fix cargo fmt --all -- --check

MarioYellowy commented:
No

Brayan-724 commented:
this is vibe coded bro 💀

gg0074x commented:
👌

ArturoAtomplay commented:
Just two questions: why are there comments in Spanish in the code, and why are there two adblock_plugin.rs files?

bryanlundberg commented:
https://github.com/user-attachments/assets/0e72e0ae-651f-4e61-af51-484f2e419e53

nglmercer committed:
Remove adblock plugin copy and tray adblock menu items

Deleted the unused 'adblock_plugin copy.rs' file and removed AdBlock-related menu items and event handling from the tray implementation. This streamlines the tray UI and eliminates redundant code.

nglmercer committed:
fix cargo fmt --all -- --check

nglmercer committed:
fix expected expression

nglmercer committed:
delete comments

2gn commented:
I almost forgot about this repo (lol). Thanks for the contribution, will review it when I have time.

nglmercer committed:
refactor(adblock): optimize Tauri adblock implementation with performance improvements

- Replace Mutex with RwLock in AdBlockState for better concurrent read performance
- Minify adblock.js for reduced bundle size and faster loading
- Streamline adblock_plugin.rs with simplified cache handling and error management
- Consolidate filter fetching with improved cache validation and timeout controls
- Refactor main.rs window management with better error handling and code organization
